shtdlib.sh

```diff
@@ -2059,6 +2133,6 @@ function test_shtdlib {
 }
 
 # Test bash version
-if compare_versions "${BASH_VERSION}" "4" ; then
+if "${bash_pre_v4}" ; then
```

Comment:
Don't necessarily need the double quotes if its a boolean

Reply:
Not in this case, true, I'll drop them.

Reply:
There is a difference in not using quotes when assigning a boolean to a variable vs. when evaluating the contents of a variable.

While defining it with quotes is redundant I prefer evaluating with quotes since I don't know the contents and I definitely don't want accidental word splitting when evaluating.

Reply:
Fair enough